File: Resource/51900286-51900712-51900718-51900699/source demo/Window Form/51900286-51900712-51900718-51900699/RestaurentDTO/NguyenVatLieu.cs
using System.Data;

namespace RestaurentDTO
{
    public class NguyenVatLieu
    {
        private string _maNguyenVatLieu;
        private string _tenNguyenVatLieu;
        private double _donGia;
        private string _donVi;
        private int _soLuongTonKho;
        private string _tinhTrang;

        public NguyenVatLieu(string maNguyenVatLieu, string tenNguyenVatLieu, double donGia, string donVi, int soLuongTonKho, string tinhTrang)
        {
            MaNguyenVatLieu = maNguyenVatLieu;
            TenNguyenVatLieu = tenNguyenVatLieu;
            DonGia = donGia;
            DonVi = donVi;
            SoLuongTonKho = soLuongTonKho;
            TinhTrang = tinhTrang;
        }

        public NguyenVatLieu(DataRow dataRow)
        {
            MaNguyenVatLieu = dataRow["MaNguyenVatLieu"].ToString();
            TenNguyenVatLieu = dataRow["TenNguyenVatLieu"].ToString();
            DonGia = (double)dataRow["DonGia"];
            DonVi = dataRow["DonVi"].ToString();
            SoLuongTonKho = (int)dataRow["SoLuongTonKho"];
            TinhTrang = dataRow["TinhTrang"].ToString();
        }

        public string MaNguyenVatLieu { get => _maNguyenVatLieu; set => _maNguyenVatLieu = value; }
        public string TenNguyenVatLieu { get => _tenNguyenVatLieu; set => _tenNguyenVatLieu = value; }
        public double DonGia { get => _donGia; set => _donGia = value; }
        public string DonVi { get => _donVi; set => _donVi = value; }
        public string TinhTrang { get => _tinhTrang; set => _tinhTrang = value; }
        public int SoLuongTonKho { get => _soLuongTonKho; set => _soLuongTonKho = value; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/NguyenVatLieu.java
package tdtu.advanced.java.thinh68.models;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name="nguyen_vat_lieu")
public class NguyenVatLieu {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    @Column(name = "ma_nguyen_vat_lieu")
    private Long maNguyenVatLieu;
    
    @Column(name = "ten_nguyen_vat_lieu")
    private String tenNguyenVatLieu;
    
    @Column(name = "don_gia")
    private double donGia;
    
    @Column(name = "don_vi")
    private String donVi;
    
    @Column(name = "tinh_trang")
    private String tinhTrang;
    
    @Column(name = "so_luong_ton_kho")
    private int soLuongTonKho;
    
    @OneToMany(mappedBy="nguyenVatLieu")
    private List<ChiTietPhieuNhap> nguyenVatLieu_ChiTietPhieuNhaps;
}